### DISCUSSION TITLE
Stitch images from different infrared cameras

### DISCUSSION THREAD

**Discussion by thilnguy** [OpenStitching/stitching#191] [Q&A]
Hello, I need your help.
I have 3 panorama images created by cam1 and cam2, when camera move from 0_110 and 70_180 degree,
cam1_0_110.jpg
![cam1_0_110_pano](https://github.com/OpenStitching/stitching/assets/48985195/86a29c8c-434d-43aa-abfe-987f51079ac8)
cam2_0_110.jpg
![cam2_0_110_pano](https://github.com/OpenStitching/stitching/assets/48985195/d1331f5a-74f0-4225-b825-892ad8d3dcec)
cam1_70_180.jpg
![cam1_70_180_pano](https://github.com/OpenStitching/stitching/assets/48985195/f7dcb604-0544-4494-b1fc-308311a33ce6)

If i stitch cam1_0_110.jpg and cam1_70_180.jpg, the stitching package works perfectly with (detection: sift, threshold: 0.2), 
![cam1_0_180_pano](https://github.com/OpenStitching/stitching/assets/48985195/6f9e3249-40f3-444d-a8bf-e024d2ce9e08)

but I try to stitch cam2_0_110.jpg and cam1_70_180.jpg, it fails.
![Screenshot 2024-03-15 at 10 28 36](https://github.com/OpenStitching/stitching/assets/48985195/2d643c10-8382-4b64-8bc5-b211f566ed30)
Could you please help me which configuration I need to set up to make it works?
cam1_0_110.jpg and cam2_0_110.jpg look similar because they are setup in the same place.
Thank you.

**Comment by lukasalexanderweber** [comment]
the algorithm can't find matches here. Can you tell more about your camera setup, the bands that are available and what is required as result? maybe we can estimate the camera parameters with rgb or other suitable bands and blend the multibands with the obtained registration. But for that I need more information

**Reply by thilnguy** [reply]
Thank you for your reply. What is the bands you meant? The expected result is: I can stitch cam2_0_110.jpg to cam1_70_180.jpg.

**Reply by lukasalexanderweber** [reply]
You have a multispectral camera which have [spectral bands](https://en.wikipedia.org/wiki/Spectral_band), have you?

**Reply by lukasalexanderweber** [reply]
https://en.wikipedia.org/wiki/Multispectral_imaging

**Reply by thilnguy** [reply]
Sorry, there is no such info defined in the camera specification. Only thermal focus length: Cam1: 26-105 mm while cam 2: 18mm

**Reply by lukasalexanderweber** [reply]
What I don't understand is why you want to mix the images?

**Reply by lukasalexanderweber** [reply]
Cam2 to Cam2 would work I guess

**Reply by thilnguy** [reply]
Yes, cam2 to cam 2 works well. The purpose is that I must create a wider panoramic view by combining the panorama from 2 cameras (the cameras may be the same or different models). I wonder which parameters I should tune to get it works because I tried several matches, but nothing works ....

**Reply by lukasalexanderweber** [reply]
mhh ok, I've already tried some settings and it don't work. I'm still a bit puzzled about your use case, why you need two cameras and don't take the pictures one after the other.

**Reply by thilnguy** [reply]
Because I need to build the panorama with lowest latency and in real use case, some zones in target panorama can be only covered by 1 camera, not by another and versa. So, I must need 2 cameras

**Reply by bulemagic** [reply]
have your ploblems been solved?I also meet this ploblems?Could you give me help?
